### DISCUSSION TITLE
Color changing like in Power Shell

### DISCUSSION THREAD

**Discussion by tiago138** [JanDeDobbeleer/oh-my-posh#1146] [Troubleshoot]
In Powershell depending on what you are typing the last character of the prompt will became red there is a way to keep this behavior with oh-my-posh?

**Comment by JanDeDobbeleer** [comment]
@Tiago138 yes, have a look at the [exit code](https://ohmyposh.dev/docs/exit) segment. This will be 2 forward arrows that always show in green and color red when there's an error:

```json
{
    "type": "exit",
     "style": "plain",
     "foreground": "#22da6e",
     "properties": {
            "prefix": "\ue285\ue285",
            "display_exit_code": false,
            "always_enabled": true,
            "error_color": "#ef5350"
    }
}
```

**Reply by tiago138** [reply]
Thanks, it was kind my mistake for not understanding how exit code works. Then I'm guessing it don't have an option for it to get red while I'm typing, it will only show the error after I press enter.

**Reply by JanDeDobbeleer** [reply]
@Tiago138 ah. I fail to see in which case this happens though? The default prompt doesn't do that, is there something I'm missing?

**Reply by tiago138** [reply]
![Terminal](https://user-images.githubusercontent.com/88953821/139848741-4e9d1bae-6942-48d2-9294-7792072ce119.JPG)

If you give a look at the terminal when I type "5 +" it already gets red showing that it is incomplete, it happens in other situations as well.

**Reply by JanDeDobbeleer** [reply]
I can use that example, but, I have no idea what's doing that (was looking at [PSReadline](https://docs.microsoft.com/en-us/dotnet/api/microsoft.powershell.psconsolereadline.selfinsert?view=powershellsdk-1.1.0), but no immediate pointers there).

**Reply by tiago138** [reply]
I was playing around with the PowerShell_profile.ps1 to see if I find what configuration does it but if you configure it like this it will still work, so I have no idea.
function prompt {  
    ">"  
}

**Reply by tiago138** [reply]
It seems it is the [ParserError](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing?view=powershell-7.1) that gives this behavior.

**Reply by JanDeDobbeleer** [reply]
@Tiago138 have a look at [line error](https://ohmyposh.dev/docs/config-line-error)

**Reply by tiago138** [reply]
lol, you implemented it!!!
Do I need to install PSReadLine? I did not install it and it seems to be working fine.
![terminal](https://user-images.githubusercontent.com/88953821/155905072-886429fe-daf4-4341-87d6-1bc329e7d76a.gif)

**Reply by JanDeDobbeleer** [reply]
No, it ships out of the box.

**Reply by tiago138** [reply]
Ok, thanks.
